core/src/kura.rs

```diff
@@ -119,7 +116,12 @@ impl Kura {
         {
             // Error here indicates the thread is already stopped, which is fine.
             let _res = shutdown_sender.send(());
-            thread_handle.join().expect("able to join");
+            if let Err(error) = thread_handle.join() {
+                error!(
+                    "Kura force shutdown routine: Kura thread panicked with error: {:?}",
+                    error
+                );
```

Comment:
Are we ignoring the error here?

Reply:
This part would probably be deleted after rebasing.

Reply:
I have indeed removed this in the rebased branch I gave to aleksandr.